src/v0/destinations/hs/util.js

```diff
@@ -464,76 +467,72 @@ const getEventAndPropertiesFromConfig = (message, destination, payload) => {
-      throw new InstrumentationError('rETL - external Id not found.');
-    }
-  } else {
-    throw new InstrumentationError('rETL - objectType or identifier type not found. ');
+const getObjectAndIdentifierType = (firstMessage) => {
+  const { objectType, identifierType } = getDestinationExternalIDInfoForRetl(
+    firstMessage,
+    DESTINATION,
+  );
+  if (!objectType || !identifierType) {
+    throw new InstrumentationError('rETL - external Id not found.');
   }
-  inputs.map(async (input) => {
+  return { objectType, identifierType };
+};
+
+/**
+ * Returns values for search api call
+ * @param {*} inputs
+ * @returns
+ */
+const extractIDsForSearchAPI = (inputs) => {
+  const values = inputs.map((input) => {
     const { message } = input;
     const { destinationExternalId } = getDestinationExternalIDInfoForRetl(message, DESTINATION);
-    values.push(destinationExternalId.toString().toLowerCase());
+    return destinationExternalId.toString().toLowerCase();
   });
 
-  values = Array.from(new Set(values));
-  const requestData = {
-    filterGroups: [
-      {
-        filters: [
-          {
-            propertyName: identifierType,
-            values,
-            operator: 'IN',
-          },
-        ],
-      },
-    ],
-    properties: [identifierType],
-    limit: SEARCH_LIMIT_VALUE,
-    after: 0,
-  };
+  return Array.from(new Set(values));
+};
 
-  const requestOptions = {
-    headers: {
-      'Content-Type': JSON_MIME_TYPE,
-      Authorization: `Bearer ${Config.accessToken}`,
-    },
-  };
-  let checkAfter = 1; // variable to keep checking if we have more results
+/**
+ * Returns hubspot records
+ * @param {*} data
+ * @param {*} requestOptions
+ * @param {*} objectType
+ * @param {*} identifierType
+ * @param {*} destination
+ * @returns
+ */
+const performHubSpotSearch = async (
+  rqdata,
+  requestOptions,
+  objectType,
+  identifierType,
+  destination,
+) => {
+  let checkAfter = 1;
+  const searchResults = [];
+  const requestData = rqdata;
+  const { Config } = destination;
 
-  /* eslint-disable no-await-in-loop */
+  const endpoint = IDENTIFY_CRM_SEARCH_ALL_OBJECTS.replace(':objectType', objectType);
+  const endpointPath = `objects/:objectType/search`;
+
+  const url =
+    Config.authorizationType === 'newPrivateAppApi'
+      ? endpoint
+      : `${endpoint}?hapikey=${Config.apiKey}`;
 
   /* *
    * This is needed for processing paginated response when searching hubspot.
    * we can't avoid await in loop as response to the request contains the pagination details
    * */
 
   while (checkAfter) {
-    const endpoint = IDENTIFY_CRM_SEARCH_ALL_OBJECTS.replace(':objectType', objectType);
-    const endpointPath = `objects/:objectType/search`;
-
-    const url =
-      Config.authorizationType === 'newPrivateAppApi'
-        ? endpoint
-        : `${endpoint}?hapikey=${Config.apiKey}`;
-    searchResponse =
+    const searchResponse =
       Config.authorizationType === 'newPrivateAppApi'
```

Comment:
const httpPOST = async (url, data, options, statTags = {})  has this signature but 
```
await httpPOST(url, requestData, {
            destType: 'hs',
            feature: 'transformation',
            endpointPath,
          });`

```
          code is wrong, should write like:
```
await httpPOST(url, requestData, {}, {
            destType: 'hs',
            feature: 'transformation',
            endpointPath,
          });
```

Reply:
addressed